lib/mpesa_stk/query_payment.rb

```diff
@@ -0,0 +1,53 @@
+require "mpesa_stk/access_token"
+
+module MpesaStk
+  class QueryPayment
+    class << self
+      def call(checkout_request_id)
+        new(checkout_request_id).query_payment
+      end
+    end
+
+    attr_reader :token, :checkout_request_id
+
+    def initialize checkout_request_id
+      @token = MpesaStk::AccessToken.call
+      @checkout_request_id = checkout_request_id
+    end
+
+    def query_payment
+      response = HTTParty.post(url, headers: headers, body: body)
+      JSON.parse(response.body)
+    end
+
+    private
+      def url
+        "#{ENV['base_url']}#{ENV['query_payment_url']}"
+      end
+
+      def headers
+        {
+          "Authorization" => "Bearer #{token}",
+          "Content-Type" => "application/json"
+        }
+      end
+
+      def body
+        {
```

Comment:
We could make it a bit robust since a couple of products are using the same endpoint

Reply:
I have added the namespace as requested.